docs/ipv6-design-doc.md

```diff
@@ -105,35 +114,66 @@ Some attempts have been made to add more robust IPv6 support:
+        discovered by the `usev4` strategy, then the address from
+        `usev4` is used and a warning is logged.
+      * If `use` and `usev6` are both set and the `use` strategy
+        discovers an IPv6 address that differs from the address
+        discovered by the `usev6` strategy, then the address from
+        `usev6` is used and a warning is logged.
+  * If `usev4` (`usev6`) is not set:
+      * If `ipv4` (`usev6`) is set, ddclient acts as if `usev4`
+        (`usev6`) was set to `ipv4` (`ipv6`).
+      * Otherwise, if `ifv4` (`ifv6`) is set, ddclient acts as if
+        `usev4` (`usev6`) was set to `ifv4` (`ifv6`).
+      * Otherwise, if `cmdv4` (`cmdv6`) is set, ddclient acts as if
+        `usev4` (`usev6`) was set to `cmdv4` (`cmdv6`).
+      * Otherwise, if `fwv4` (`fwv6`) is set, ddclient acts as if
+        `usev4` (`usev6`) was set to `fwv4` (`fwv6`).
+      * Otherwise, `usev4` (`usev6`) remains unset.
 
 ### Internal API changes
 
-TODO: Describe changes to the update protocol callback API.
+Add two new entries to the `$config{$host}` hash:
+
+  * `$config{$host}{'wantipv4'}` is set to:
+      * If `usev4` is enabled, the IPv4 address discovered by the
+        `usev4` strategy.
+      * Otherwise, if `use` is enabled and the `use` strategy
+        discovered an IPv4 address, the IPv4 address discovered by the
+        `use` strategy.
+      * Otherwise, `undef`.
+  * `$config{$host}{`wantipv6`}` is set to:
+      * If `usev6` is enabled, the IPv6 address discovered by the
+        `usev6` strategy.
+      * Otherwise, if `use` is enabled and the `use` strategy
+        discovered an IPv6 address, the IPv6 address discovered by the
+        `use` strategy.
+      * Otherwise, `undef`.
+
+Deprecate the existing `$config{$host}{'wantip'}` entry, to be removed
+after all update protocol callbacks have been updated to use the above
+new entries. In the meantime, this entry's value depends on which of
+`use`, `usev4`, and `usev6` is enabled, and what type of IP address is
+discovered by the `use` strategy (if enabled), according to the
+following table:
+
+| `use` | `usev4` | `usev6` | resulting value |
+| :---: | :---: | :---: | :--- |
+| ✔(IPv4) | ✖ | ✖ | the IPv4 address discovered by the `use` strategy |
+| ✔(IPv6) | ✖ | ✖ | the IPv6 address discovered by the `use` strategy |
+| ✖ | ✔ | ✖ | the IPv4 address discovered by the `usev4` strategy |
+| ✖ | ✖ | ✔ | the IPv6 address discovered by the `usev6` strategy |
+| ✔(IPv4) | ✔ | ✖ | the IPv4 address discovered by the `usev4` strategy (and log a warning if it doesn't match the IPv4 address found by the `use` strategy) |
+| ✔(IPv6) | ✔ | ✖ | the IPv6 address discovered by the `use` strategy |
+| ✔(IPv4) | ✖ | ✔ | the IPv4 address discovered by the `use` strategy |
+| ✔(IPv6) | ✖ | ✔ | the IPv6 address discovered by the `usev6` strategy (and log a warning if it doesn't match the IPv6 address found by the `use` strategy) |
```

Comment:
Why bother testing if `use` equals `usev4` or `usev6` and issue a warning?  We should issue the warning whether it matches or not (assuming it is of the same version) so that user is encouraged to remove `use` and replace with the intended replacement.

Reply:
In fact we should always issue a warning if `use` is set whether or not `usev4` `usev6` are set or match or not.  Setting `use` is always reason for a warning.

Reply:
`use` will always print a warning because it's deprecated. What I mean here is that a second warning would be printed because the IP address discovered by `use` is discarded in favor of the different address discovered by `usev4` or `usev6`.